cylc/flow/loggingutil.py

```diff
@@ -155,10 +154,19 @@ def __init__(
         self.no_detach = no_detach
         self.formatter = CylcLogFormatter(timestamp=timestamp)
         # Header records get appended to when calling
-        # LOG.info(extra=RotatingLogFileHandler.[rollover_]header_extra)
+        # `LOG.info(extra=RotatingLogFileHandler.[rollover_]header_extra)`:
         self.header_records: List[logging.LogRecord] = []
         self.restart_num = restart_num
         self.log_num: Optional[int] = None  # null value until log file created
+        # Get & cache properties from global config (note: we should not access
+        # the global config object during emit as it can have side effects)
```

Comment:
Can you elaborate on this a bit? The side effect is to expand the sparse config, and why is that bad in this context?

Reply:
(Approved the PR anyway, as comment is obviously non-functional, but personally I'd like a bit more info here)

Reply:
```suggestion
        # Get & cache properties from global config (note: we should not access
        # the global config object while running the `emit` method 
        # becuase ...)
```

Reply:
> The side effect is to expand the sparse config, and why is that bad in this context?

IMO that is self-explanatory - logging is a fundamental operation and should not have the possibility of any real side effects at all

Reply:
```suggestion
        # Get & cache properties from global config (note: we should not access
        # the global config object during the emit method as it can have side effects)
```